### PR TITLE
105 experiment with python fire

### PR DESCRIPTION
**Pull request by tschm**

### CODE REVIEW

File: cvx/cli/weather_fire.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        if metric in r.json()["current_weather"]:
+            x = r.json()["current_weather"][metric]
+            return x
+        else:
+            raise ValueError("Metric not supported!")
+    else:
+        raise ConnectionError("Open-Meteo is down!")
+
+
+def fff():  # pragma: no cover
```
Comment (phschiele) [inline cvx/cli/weather_fire.py:33]: I usually put the `Fire` call into a `if __name__ == "__main__` clause.

**Review by phschiele — APPROVED** [review]
@tschm There are some relevant changes in this PR, e.g. changes to serialization and deserialization. 
Some files appear to be test files (e.g. weather CLI). Do you want to remove them before or after merging this PR?
Approving the PR in case you want to do it later.

File: cvx/cli/weather_fire.py (hunk 1 above)
Comment (tschm) [reply cvx/cli/weather_fire.py:33]: it works better with pyproject.toml if it is hidden in a function which should have a more appropriate name though. I call it main() now... I can then do poetry run weather-fire temperature

**Comment by tschm** [comment]
> @tschm There are some relevant changes in this PR, e.g. changes to serialization and deserialization.
> Some files appear to be test files (e.g. weather CLI). Do you want to remove them before or after merging this PR?
> Approving the PR in case you want to do it later.

Yes, weather-app will disappear soon. I copied it from the web to experiment with the technology.

File: cvx/cli/weather_fire.py (hunk 1 above)
Comment (tschm) [reply cvx/cli/weather_fire.py]: https://github.com/cvxgrp/cvxmarkowitz/blob/96da221368b7331025f9dc84b3d46bbfa20ca937/pyproject.toml#L53

Comment (tschm) [reply cvx/cli/weather_fire.py]: [tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026

Comment (tschm) [reply cvx/cli/weather_fire.py]: sorry, I didn't try to put any explicit emphasis on the link :-)

Comment (tschm) [reply cvx/cli/weather_fire.py]: `[tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026
`

Comment (tschm) [reply cvx/cli/weather_fire.py]:
```
[tool.poetry.scripts]
weather-fire = "cvx.cli.weather_fire:main"

# Relevant documentation:
# https://medium.com/clarityai-engineering/how-to-create-and-distribute-a-minimalist-cli-tool-with-python-poetry-click-and-pipx-c0580af4c026
```

**Pull request merged**